File: src/main/java/dominio/metodologias/Condicion.java
package dominio.metodologias;

import javax.persistence.*;

@MappedSuperclass
public abstract class Condicion {
	@Id 
	@GeneratedValue
	private Long id;
	
	@OneToOne(cascade = CascadeType.PERSIST, fetch = FetchType.EAGER)
	protected OperandoCondicion operando;
	@Enumerated
	protected OperacionRelacional operacionRelacional;
	
	public OperandoCondicion getOperando() {
		return operando;
	}
}

File: src/main/java/dominio/metodologias/Condicion.kt
package dominio.metodologias

import javax.persistence.*

@MappedSuperclass
abstract class Condicion(@OneToOne(cascade = [CascadeType.PERSIST], fetch = FetchType.EAGER)
                         val operando: OperandoCondicion,
                         @Enumerated
                         protected val operacionRelacional: OperacionRelacional
) {
    @Id
    @GeneratedValue
    private val id: Long? = null
}